test/unit/mdc-tabs/mdc-tab-bar.test.js

```diff
@@ -0,0 +1,328 @@
+
+  assert.isArray(component.tabs);
+
+  component.tabs.forEach((tab) => {
+    assert.instanceOf(tab, MockTab);
+  });
+});
+
+test('#get activeTab returns active tab', () => {
+  const {component} = setupTest();
+  const targetIndex = 1;
+  const tab = component.tabs[targetIndex];
+
+  tab.isActive = true;
+  assert.equal(tab, component.activeTab);
+});
+
+test('#set activeTab proxies to foundation_.switchToTabAtIndex', () => {
+  const raf = createMockRaf();
+  const {component} = setupTest();
+  const tab = new MockTab();
+  component.tabs.push(tab);
+
+  component.activeTab = tab;
+  raf.flush();
+
+  assert.isTrue(tab.isActive);
+  raf.restore();
+});
+
+test('#get activeTabIndex returns active tab', () => {
+  const {component} = setupTest();
+  const targetIndex = 0;
+  const tab = component.tabs[targetIndex];
+
+  tab.isActive = true;
+  assert.equal(component.tabs.indexOf(tab), component.activeTabIndex);
+});
+
+test('#set activeTabIndex proxies to foundation_.switchToTabAtIndex', () => {
+  const raf = createMockRaf();
+  const {component} = setupTest();
+  const tab = new MockTab();
+  component.tabs.push(tab);
+
+  component.activeTabIndex = component.tabs.indexOf(tab);
+  raf.flush();
+
+  assert.isTrue(tab.isActive);
+  raf.restore();
+});
+
+test('adapter#addClass adds a class to the root element', () => {
+  const {root, component} = setupTest();
+
+  component.getDefaultFoundation().adapter_.addClass('foo');
+  assert.isOk(root.classList.contains('foo'));
+});
+
+test('adapter#removeClass removes a class from the root element', () => {
+  const {root, component} = setupTest();
+  root.classList.add('foo');
+
+  component.getDefaultFoundation().adapter_.removeClass('foo');
+  assert.isNotOk(root.classList.contains('foo'));
+});
+
+test('adapter#bindOnMDCTabSelectedEvent adds event listener for MDCTab:selected on ' +
+  'the component', () => {
+  const {component, root} = setupTest();
+  const adapter = component.getDefaultFoundation().adapter_;
+  const tab = new MockTab();
+  component.tabs.push(tab);
+  const raf = createMockRaf();
+
+  adapter.bindOnMDCTabSelectedEvent();
+  domEvents.emit(root, 'MDCTab:selected', {detail: {tab}});
+
+  raf.flush();
+
+  assert.isTrue(tab.isActive);
+  raf.restore();
+});
+
+test('on MDCTab:selected if tab is not in tab bar, throw Error', () => {
+  const {component} = setupTest();
+  const tab = new MockTab();
+
+  // This is purely to make sure that bindOnMDCTabSelectedEvent() is executed
+  // and does not throw.
+  component.getDefaultFoundation().adapter_.bindOnMDCTabSelectedEvent();
+
+  // NOTE: Because of this issue: https://bugs.chromium.org/p/chromium/issues/detail?id=239648#
+  // Chai cannot observe an error thrown as a result of an event being dispatched.
+  const evtObj = {
+    detail: {tab},
+  };
+  assert.throws(() => component.tabSelectedHandler_(evtObj));
+});
+
+test('adapter#unbindOnMDCTabSelectedEvent removes listener from component', () => {
```

Comment:
I don't think this is testing what you think.

You're listening for `MDCTab:selected` on the component but firing it on `tab.root`. Are you sure the event is propagating up to the component level? Because if so that handler should be called, right?

You may instead want to use a similar approach above, and check that `detail.tab` is _not_ active after the event is emitted on the component's root element.